model/src/main/java/org/projectnessie/model/Branch.java

```diff
@@ -57,24 +52,15 @@ static Branch of(String name, String hash) {
   }
 
   /**
-   * Returns a map of metadata properties that describe this branch. Note that these properties
-   * <b>can be added</b> by the server when a {@link Branch} instance is returned. A returned {@link
-   * Branch} instance can then have the following keys:
+   * Returns a {@link ReferenceMetadata} instance that contains additional metadata about this
+   * branch. Note that this is <b>only added</b> by the server when <b>explicitly</b> requested by
+   * the client.
    *
-   * <ul>
-   *   <li>{@value Branch#NUM_COMMITS_AHEAD}: number of commits ahead of the default branch
-   *   <li>{@value Branch#NUM_COMMITS_BEHIND}: number of commits behind the default branch
-   *   <li>{@value Branch#HEAD_COMMIT_META}: the commit metadata of the HEAD commit
-   *   <li>{@value Branch#COMMON_ANCESTOR_HASH}: the hash of the common ancestor in relation to the
-   *       default branch.
-   * </ul>
-   *
-   * @return A map of metadata properties that describe this branch. Note that these properties
-   *     <b>can be added</b> by the server when a {@link Branch} instance is returned.
+   * @return A {@link ReferenceMetadata} instance that contains additional metadata about this
+   *     branch. Note that this is <b>only added</b> by the server when <b>explicitly</b> requested
+   *     by the client.
    */
-  @Value.Default
-  @JsonInclude(Include.NON_EMPTY)
-  default Map<String, Object> metadataProperties() {
-    return Collections.emptyMap();
-  }
+  @JsonInclude(Include.NON_NULL)
+  @Nullable
+  ReferenceMetadata metadata();
```

Comment:
I think, this should live in `Reference`

Reply:
fixed